torchvision/csrc/io/decoder/__init__.py

```diff
@@ -0,0 +1,208 @@
+import torch
+from typing import Any, Dict, Iterator
+
+from ._video_opt import (
+    Timebase,
+    VideoMetaData,
+    _HAS_VIDEO_OPT,
+    _probe_video_from_file,
+    _probe_video_from_memory,
+    _read_video_from_file,
+    _read_video_from_memory,
+    _read_video_timestamps_from_file,
+    _read_video_timestamps_from_memory,
+)
+from .video import (
+    read_video,
+    read_video_timestamps,
+    write_video,
+)
+from .image import (
+    ImageReadMode,
+    decode_image,
+    decode_jpeg,
+    decode_png,
+    encode_jpeg,
+    encode_png,
+    read_file,
+    read_image,
+    write_file,
+    write_jpeg,
+    write_png,
+)
+
+
+if _HAS_VIDEO_OPT:
+
+    def _has_video_opt() -> bool:
+        return True
+
+
+else:
+
+    def _has_video_opt() -> bool:
+        return False
+
+
+class VideoReader:
```

Comment:
Isn't this already defined in [`torchvision/io/__init__.py`](https://github.com/pytorch/vision/blob/972ca657416c5896ba6e0f5fe7c0d0f3e99e1087/torchvision/io/__init__.py#L47).